File: 프로그래머스/2/12946. 하노이의 탑/하노이의 탑.js
function solution(n) {
    let answer = [];
    
    function hanoi(n, from , to){
        if(n === 1) return answer.push([from,to])
        
        const empty = 6 - from - to 
        return hanoi(n-1,from, empty) +hanoi(1,from, to)+hanoi(n-1,empty,to)
    }
    
    hanoi(n,1,3)
    return answer;
}

File: 프로그래머스/2/12946. 하노이의 탑/하노이의 탑.java
import java.util.*;
class Solution {
    public List<int[]> solution(int n) {
        List<int[]> answer = new ArrayList<>();
        hanoi(n,1,3,answer);
        return answer;
    }
    
    public void hanoi(int n , int from , int to, List<int[]> answer){
        if(n == 1) {answer.add(new int[]{from,to});
                    return;};
        int temp = 6 - (from + to);
        hanoi(n-1,from,temp,answer );
        answer.add(new int[]{from,to});
        hanoi(n-1, temp, to, answer);
        
    }
}